analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -146,7 +156,32 @@ private fun resolveExternalDocumentReference(
             ?: throw IllegalArgumentException("The local file URI '$uri' does not point to an existing file. ")
     }
 
-    throw IllegalArgumentException("Only URIs to local files are supported for now, but '$uri' is none.")
+    return requestSpdxDocument(uri)
+}
+
+/**
+ * Returns [File] from a given [URI] if it can be found.
+ * @throws DownloadException when file can not be downloaded.
+ */
+private fun requestSpdxDocument(uri: URI): File {
+    val request = Request.Builder().get().url(uri.toURL()).build()
+
+    OkHttpClientHelper.execute(request).use { response ->
```

Comment:
By now, there's the `OkHttpClientHelper.downloadText()` helper method that you could leverage. Is there a way we could avoid writing to a temporary file?

Reply:
> Is there a way we could avoid writing to a temporary file?

Maybe by changing `resolveExternalDocumentReference` to already return an `SpdxDocument` instead of a `File`?

Reply:
It might be difficult to get around the temporary file. As of now the `SpdxModelMapper` only takes files: `SpdxModelMapper.read<SpdxDocument>(file)`

Reply:
But `SpdxModelMapper` also has `fromYaml` / `fromJson`.